### PR TITLE
ENG-966 / ENG-967 - AI Hackstack create mode in teacher dashboard

### PR DESCRIPTION
**Pull request by adamkecskes**
- New icon for 'use' mode scenarios.
 - Show them completed if `isReadyToReview` is checked.

AI PR: https://github.com/codecombat/ai/pull/264
Backend PR: https://github.com/codecombat/codecombat-server/pull/1020
Notification_Center

## Summary by CodeRabbit

- **New Features**
  - Added localization support with new phrases to enhance user engagement.
  - Enhanced project status display with contextual information based on the current scenario mode in the teacher dashboard.
  
- **Improvements**
  - Improved conditional rendering in components for a more dynamic user interface.
  - Refined logic for adjusting learning goals based on project scenarios.

- **Bug Fixes**
  - Resolved issues related to icon rendering conditions in the content display component.

### CODE REVIEW

File: ozaria/site/components/teacher-dashboard/Panel/components/AiProject.vue [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
-    <p>Progress: {{ progress }}%</p>
-    <p v-if="failedAttempts">
-      {{ $t('teacher_dashboard.failed_attempts') }}: {{ failedAttempts }}
-      <br><span class="subtext">{{ $t('teacher_dashboard.failed_attempts_subtext') }}</span>
-    </p>
-    <p v-else>
-      {{ $t('teacher_dashboard.no_failed_attempts') }}
+
+    <p v-if="mode === 'use'">
+      {{ aiProject.isReadyToReview ? $t('teacher_dashboard.ready_to_review') : $t('teacher.in_progress') }}
```
Comment (mrfinch) [inline ozaria/site/components/teacher-dashboard/Panel/components/AiProject.vue:6]: seems like u missed adding i18n for this one in en.js

File: ozaria/site/components/teacher-dashboard/Panel/components/AiProject.vue [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
     </p>
+    <div v-else>
+      <p>
+        Progress: {{ progress }}%
```
Comment (mrfinch) [inline ozaria/site/components/teacher-dashboard/Panel/components/AiProject.vue:10]: can we make this p bold to stand out in this section?

**Review by mrfinch — COMMENTED** [review]
- we should remove 'Learning Goal' section if it's empty
- why does use-gpt4 come twice? we should show it only once?

Screenshot 2024-08-14 at 10 20 58 AM

**Comment by adamkecskes** [comment]
@mrfinch 
> why does use-gpt4 come twice? we should show it only once?
ENG-966___ENG-967_-_AI_Hackstack_create_mode_in_teacher_dashboard_by_adamkecskes_·_Pull_Request__7630_·_codecombat_codecombat
> we should remove 'Learning Goal' section if it's empty
Removed for "Create" mode, but I've added `aiScenario.name` for "Learn" mode projects:
CodeCombat_Teacher_Dashboard

**Review by mrfinch — APPROVED** [review]

**Pull request merged**